src/components/views/messages/ReactionsRowButton.tsx

```diff
@@ -106,21 +107,48 @@ export default class ReactionsRowButton extends React.PureComponent<IProps, ISta
 
         const room = this.context.getRoom(mxEvent.getRoomId());
         let label: string | undefined;
+        let customReactionName = "";
         if (room) {
             const senders: string[] = [];
             for (const reactionEvent of reactionEvents) {
                 const member = room.getMember(reactionEvent.getSender()!);
                 senders.push(member?.name || reactionEvent.getSender()!);
+                if (reactionEvent.event.content?.["com.beeper.reaction.shortcode"]) {
+                    customReactionName = reactionEvent.event.content?.["com.beeper.reaction.shortcode"];
+                }
             }
 
             const reactors = formatCommaSeparatedList(senders, 6);
             if (content) {
-                label = _t("%(reactors)s reacted with %(content)s", { reactors, content });
+                label = _t("%(reactors)s reacted with %(content)s", {
+                    reactors,
+                    content: customReactionName || content,
+                });
             } else {
                 label = reactors;
             }
         }
 
+        let reactionContent = (
+            <span className="mx_ReactionsRowButton_content" aria-hidden="true">
+                {content}
+            </span>
+        );
+        if (content.startsWith("mxc://")) {
+            const imageSrc = mediaFromMxc(content).srcHttp;
+            if (imageSrc) {
+                reactionContent = (
+                    <img
+                        className="mx_ReactionsRowButton_content"
+                        alt={customReactionName || content}
+                        src={imageSrc}
+                        width="16"
+                        height="16"
+                    />
+                );
+            }
+        }
```

Comment:
Could this be an if-else statement?

Reply:
The reason I did it this way is that there are two necessary conditions for rendering as an image:

1. Does it start with `mxc://`?
2. If so, then does getting the media src using `mediaFromMxc(content).srcHttp` work?

Both of these have to be true in order for it to make sense to render as an image.

Since JavaScript doesn't have any way to declare a variable that can be used within the block within the `if` condition itself, I thought this would be the best option, since otherwise we would need to duplicate the `span` code in the `else` of both the inner and outer `if`s.

If this was Python for example, I think that doing:

```python
if content.startswith("mxc://") and (image_src := media_from_mxc(content).src_http):
    reaction_content = (<img...>)
else:
    reaction_content = (<span...>)
```

Reply:
Ah, yeah, you're right!